src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/RdsHostUtils.java

```diff
@@ -227,19 +227,46 @@ public Map<String, String> getExplicitlySetProperties(final PropertySet property
   }
 
   public int getHostIndex(List<HostInfo> hostList, HostInfo host) {
-    if (host == null || Util.isNullOrEmpty(hostList)) {
+    if (host == null) {
```

Comment:
NIT: you can leave the hostList check here, that returns immediately (even though it will return in the next stack call)

Reply:
Yeah good point, probably makes more sense to leave it in